include/micm/solver/backward_euler.inl

```diff
@@ -157,29 +157,22 @@ namespace micm
                       (std::abs(*forcing_iter) <= parameters_.relative_tolerance_ * std::abs(*yn1_iter));
           ++forcing_iter, ++yn1_iter, ++abs_tol_iter;
         } while (converged && forcing_iter != forcing.end());
-
-        if (!converged)
-        {
-          std::cout << "failed to converge within the newton iteration\n";
-        }
       } while (!converged && iterations < max_iter);
 
       if (!converged)
       {
-        std::cout << "failed to converge\n";
         n_successful_integrations = 0;
 
         if (n_convergence_failures >= time_step_reductions.size())
         {
-          // we have failed to converge, accept the solution
-          // TODO: continue on with the current solution to get the full solution
-          n_convergence_failures = 0;
-          // give_up = true;
-          t += H;
-          throw std::system_error(make_error_code(MicmBackwardEulerErrc::FailedToConverge), "Failed to converge");
-        };
-
-        H *= time_step_reductions[n_convergence_failures++];
+          std::cerr << "Failed to converge too many times in a row. Accepting the current integration and continuing on.\n";
+          break;
```

Comment:
according to https://github.com/ESCOMP/CAM/blob/cam_development/src/chemistry/pp_trop_strat_mam4_vbs/mo_imp_sol.F90#L385-L397, we should advance `t` to `t+H` here. The comments are confusing to me as we only advance the time step, but do not accept the changed solution (i.e., the `base_sol` in the F90 code is not updated).

Reply:
ah good catch. I thought I was missing something but struggled to understand the linked code